File: lang/go/test/muse/data_structures/ArrayQueue/ArrayQueueTest.go
package main

import (
    "muse/data_structures/ArrayQueue"
    "muse/util/TestRunner"
)

func testArrayQueue() bool {
    size := 8192

    queue := ArrayQueue.New(0)

    for i := 1; i <= size; i++ {
        queue.Enqueue(i)
    }

    queue.Shrink()

    if queue.Size() != size {
        return false
    }

    if queue.Capacity() != size {
        return false
    }

    for i := 1; i <= size; i++ {
        if queue.Peek() != i {
            return false
        }
        queue.Dequeue()
    }

    queue.Shrink()

    if !queue.IsEmpty() {
        return false
    }

    return queue.Capacity() == 0
}

func main() {
    TestRunner.ParseTest(testArrayQueue())
}

File: lang/ruby/test/muse/data_structures/ArrayQueueTest.rb
require "muse/data_structures/ArrayQueue"
require "muse/util/TestRunner"

def testArrayQueue()
    size = 8192

    queue = ArrayQueue::ArrayQueue.new(0)

    1.upto(size) do |i|
        queue.enqueue(i)
    end

    queue.shrink()

    if queue.size() != size
        return false
    end

    if queue.capacity() != size
        return false
    end

    1.upto(size) do |i|
        if queue.peek() != i
            return false
        end
        queue.dequeue()
    end

    queue.shrink()

    unless queue.isEmpty()
        return false
    end

    queue.capacity().zero?
end

if __FILE__ == $PROGRAM_NAME
    TestRunner.parseTest(testArrayQueue())
end
